### PR TITLE
[text-box-edge] Fix computed values

### PR DESCRIPTION
**Pull request by zakariaridouh**
#### 398e5e25e9f250e1a4e3f4dde3ae54dd09dbe23e
<pre>
[text-box-edge] Fix computed values
<a href="https://bugs.webkit.org/show_bug.cgi?id=263508">https://bugs.webkit.org/show_bug.cgi?id=263508</a>
<a href="https://rdar.apple.com/117643064">rdar://117643064</a>

Reviewed by Tim Nguyen.

 Combined changes:
 * LayoutTests/imported/w3c/web-platform-tests/css/css-inline/text-box-trim/inheritance-expected.txt:
 * LayoutTests/imported/w3c/web-platform-tests/css/css-inline/text-box-trim/text-box-edge-computed-expected.txt:
 * LayoutTests/imported/w3c/web-platform-tests/css/cssom/cssom-getPropertyValue-common-checks-expected.txt:
 * Source/WebCore/css/ComputedStyleExtractor.cpp:
 (WebCore::valueForTextBoxEdge):

Canonical link: <a href="https://commits.webkit.org/280413@main">https://commits.webkit.org/280413@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/41cf97ecc3263a4a5082740f2b7ff5bb8724d3b3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56527 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35854 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9000 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60138 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6968 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58654 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43477 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7159 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45793 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4870 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58557 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33720 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48789 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26648 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30500 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/6118 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/5971 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/52463 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6390 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/61821 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/437 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6498 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53054 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/439 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48846 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/52968 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12515 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/380 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31681 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32768 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33852 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32514 "Built successfully") | | |

### CODE REVIEW

File: Source/WebCore/css/ComputedStyleExtractor.cpp [hunk 1]
```diff
@@ -1268,6 +1268,9 @@ static Ref<CSSValue> valueForScrollbarGutter(const ScrollbarGutter& gutter)
 
 static Ref<CSSValueList> valueForTextBoxEdge(const TextBoxEdge& textBoxEdge)
 {
+    if (textBoxEdge.over == TextBoxEdgeType::Leading && textBoxEdge.under == TextBoxEdgeType::Leading)
+        return CSSValueList::createSpaceSeparated(createConvertingToCSSValueID(textBoxEdge.over));
```
Comment (nt1m) [inline Source/WebCore/css/ComputedStyleExtractor.cpp]:
```suggestion
    if (textBoxEdge.over == TextBoxEdgeType::Leading && textBoxEdge.under == TextBoxEdgeType::Leading)
        return createConvertingToCSSValueID(textBoxEdge.over);
```

This should also work, and you change the return type  of the function to `Ref<CSSValue>`.

File: LayoutTests/imported/w3c/web-platform-tests/css/css-inline/text-box-trim/text-box-edge-computed-expected.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 
-FAIL Property text-box-edge value 'leading' assert_equals: expected "leading" but got "leading leading"
+PASS Property text-box-edge value 'leading'
 FAIL Property text-box-edge value 'text' assert_equals: expected "text" but got "text text"
 FAIL Property text-box-edge value 'ex' assert_equals: expected "ex" but got "ex text"
 FAIL Property text-box-edge value 'cap' assert_equals: expected "cap" but got "cap text"
```
Comment (nt1m) [inline LayoutTests/imported/w3c/web-platform-tests/css/css-inline/text-box-trim/text-box-edge-computed-expected.txt]: You could fix the other ones pretty easily :) 

```
if (textBoxEdge.under == TextBoxEdgeType::Text)
    return createConvertingToCSSValueID(textBoxEdge.over);
```

**Review by nt1m — COMMENTED** [review]
You'll want to rebaseline these as well:

imported/w3c/web-platform-tests/css/css-inline/text-box-trim/inheritance.html
imported/w3c/web-platform-tests/css/cssom/cssom-getPropertyValue-common-checks.html

**Review by nt1m — APPROVED** [review]
Looks good, thanks! Please squash everything in one commit, EWS won't be able to land it otherwise.

File: Source/WebCore/css/ComputedStyleExtractor.cpp (hunk 1 above)
Comment (zakariaridouh) [reply Source/WebCore/css/ComputedStyleExtractor.cpp]: fixed :)

File: LayoutTests/imported/w3c/web-platform-tests/css/css-inline/text-box-trim/text-box-edge-computed-expected.txt (hunk 2 above)
Comment (zakariaridouh) [reply LayoutTests/imported/w3c/web-platform-tests/css/css-inline/text-box-trim/text-box-edge-computed-expected.txt]: thanks!

File: Source/WebCore/css/ComputedStyleExtractor.cpp [hunk 3]
```diff
@@ -1268,6 +1268,9 @@ static Ref<CSSValue> valueForScrollbarGutter(const ScrollbarGutter& gutter)
 
 static Ref<CSSValueList> valueForTextBoxEdge(const TextBoxEdge& textBoxEdge)
 {
+    if (textBoxEdge.over == TextBoxEdgeType::Leading && textBoxEdge.under == TextBoxEdgeType::Leading)
```
Comment (pxlcoder) [inline Source/WebCore/css/ComputedStyleExtractor.cpp:1271]: @nt1m Is there a reason this condition shouldn't simply be `textBoxEdge.over == textBoxEdge.under`?

There is another failing test which says:
```
FAIL Property text-box-edge value 'text' assert_equals: expected "text" but got "text text"
```

Comment (nt1m) [reply Source/WebCore/css/ComputedStyleExtractor.cpp:1271]: That's not correct. `ideographic ideographic` is not equivalent to `ideographic`. `ideographic` is equivalent to `ideographic text`.

However, `leading leading` is not a correct value but `leading` is.

See: https://drafts.csswg.org/css-inline-3/#text-edges

**Pull request merged**